spec/index.html

```diff
@@ -857,30 +857,212 @@ <h3>Simple N3 interpretations</h3>
+		  <li><a href="../ns/time.html">Time</a> –
+			<a href="http://www.w3.org/2000/10/swap/time#">http://www.w3.org/2000/10/swap/time#</a>.</li>
+		</ul>
+	
+	<p>A built-in definition is structured as follows:
+		<ul>
+			<li><b>schema</b>: set of <a>arguments</a> defined in terms of the <a>N3 triple elements</a> in a built-in statement for which they act as placeholders.</li>
+			<li><b>domain</b>: expected <a>domain datatypes</a> for the built-in arguments. Casting may take place if the <a>value datatype</a> does not correspond to the domain datatype.</li>
+			<li><b>inputs and outputs</b>: <a>input arguments</a> and <a>output arguments</a>.</li>
+			<li><b>summary</b>: a human-readable summary of the function of the built-in.</li>
+		</ul>
+		
+		We define these concepts in more detail below.
+	</p>
+	
+	<section id="builtins_args">
+		<h4>Input and output arguments</h4>
+		<p>An N3 <a>built-in</a> operates on its arguments. An <dfn>argument</dfn> is a placeholder that refers to an <a>N3 triple element</a> from the <dfn>built-in statement</dfn>, i.e., the <a>N3 statement</a> where the built-in acts as a predicate.</p>
+		<p>In the simplest case, there are two arguments that respectively refer to the subject and object of the built-in statement. For instance, statements such as `1 math:lessThan 2` are true if the subject is less than the object. Hence, built-in `math:lessThan` has two arguments `$a_1` and `$a_2`, which is written as follows: `$a_1 math:lessThan $a_2`. (In this section, arguments are prefixed with `$a`.)</p>
+		<p>Arguments can be also represent a "deconstruction" of the subject or object in case of lists or cited formulas. For instance, `(1 2 3) math:sum 6` calculates the sum of a list of numbers. Hence, the built-in `math:sum` has arguments `$a_1` .. `$a_n` and `$a_sum`, which is written as follows: `($a_1 .. $a_n) math:sum $a_sum`. </p>			
+		<p>In a built-in statement, if an argument refers to a concrete <a>N3 triple element</a> (i.e., either a concrete literal or a bound variable) within the built-in statement, we call its referent a built-in <dfn>input</dfn>, else, we call its referent a built-in <dfn>output</dfn>.
+		<p>In some cases, an N3 built-in will only support outputs for one argument:
+			e.g., `math:sum` only supports outputs for `$a_sum`.
+			In other cases, an N3 built-in only supports inputs for all arguments: 
+			e.g., `math:lessThan` only supports inputs for `$a_1` and `$a_2`.
+			In rarer cases, an N3 built-in can support outputs for several arguments. 
+			E.g., for `35 math:cos ?x`, the `math:cos` of `35` will be calculated; for `?x math:cos 0.57` the inverse operation (`math:acos`) for `0.57` will be calculated.
+		</p>
+		<p>The set of arguments, and whether they are allowed to have inputs or outputs, are defined per N3 built-in. An argument that is allowed to have output referents is called an <dfn>output argument</dfn>, else it is called an <dfn>input argument</dfn>.</p>
```

Comment:
```suggestion
		<p>The set of arguments, and whether they are allowed to have inputs or outputs, are defined per N3 built-in. An argument that is allowed to have output referents is called an <dfn>output argument</dfn>; otherwise, it is called an <dfn>input argument</dfn>.</p>
```